core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -312,6 +316,15 @@ private[zio] final class FiberContext[E, A](
   private[this] final def getDescriptor: Fiber.Descriptor =
     Fiber.Descriptor(fiberId, state.get.interrupted, executor)
 
+  private[this] final def getSupervising: List[Fiber[_, _]] = {
+    import collection.JavaConverters._
```

Comment:
How about using `.asScalaIterator` directly?

Reply:
If it's more efficient and hides the mutations of the set from the iterator, then yes!